aptos-move/framework/aptos-framework/sources/staking_contract.move

```diff
@@ -0,0 +1,922 @@
+        distribute_internal(staker_address, old_operator, &mut staking_contract, &mut store.distribute_events);
+
+        // For simplicity, we request commission to be paid out first. This avoids having to ensure to staker doesn't
+        // withdraw into the commission portion.
+        request_commission_internal(
+            old_operator,
+            &mut staking_contract,
+            &mut store.add_distribution_events,
+            &mut store.request_commission_events,
+        );
+
+        // Update the staking contract's commission rate and stake pool's operator.
+        stake::set_operator_with_cap(&staking_contract.owner_cap, new_operator);
+        staking_contract.commission_percentage = new_commission_percentage;
+
+        let pool_address = staking_contract.pool_address;
+        simple_map::add(staking_contracts, new_operator, staking_contract);
+        emit_event(
+            &mut store.switch_operator_events,
+            SwitchOperatorEvent { pool_address, old_operator, new_operator }
+        );
+    }
+
+    /// Allow anyone to distribute already unlocked funds. This does not affect reward compounding and therefore does
+    /// not need to be restricted to just the staker or operator.
+    public entry fun distribute(staker: address, operator: address) acquires Store {
+        assert_staking_contract_exists(staker, operator);
+        let store = borrow_global_mut<Store>(staker);
+        let staking_contract = simple_map::borrow_mut(&mut store.staking_contracts, &operator);
+        distribute_internal(staker, operator, staking_contract, &mut store.distribute_events);
+    }
+
+    /// Distribute all unlocked (inactive) funds according to distribution shares.
+    fun distribute_internal(
+        staker: address,
+        operator:address,
+        staking_contract: &mut StakingContract,
+        distribute_events: &mut EventHandle<DistributeEvent>,
+    ) {
+        let pool_address = staking_contract.pool_address;
+        let (_, inactive, _, _) = stake::get_stake(pool_address);
+        // No coins to distribute. End early.
+        if (inactive == 0) {
+            return
+        };
+        let coins = stake::withdraw_with_cap(&staking_contract.owner_cap, inactive);
+        let distribution_pool = &mut staking_contract.distribution_pool;
+        // Update the number of coins in distribution pool to match current value.
+        pool_u64::update_total_coins(distribution_pool, coin::value(&coins));
+
+        // Buy all recipients out of the distribution pool.
+        while (pool_u64::shareholders_count(distribution_pool) > 0) {
+            let recipients = pool_u64::shareholders(distribution_pool);
+            let recipient = *vector::borrow(&mut recipients, 0);
+            let current_shares = pool_u64::shares(distribution_pool, recipient);
+            let amount_to_distribute = pool_u64::redeem_shares(distribution_pool, recipient, current_shares);
```

Comment:
so we basically empty out the distribution_pool?

Reply:
Yes. This is because it's only tracking pending distributions. Once they're paid out, they're removed from the pool